### DISCUSSION TITLE
v3.0.0-rc.2

### DISCUSSION THREAD

**Discussion by carhartl** [js-cookie/js-cookie#694] [General]
* Improve module setup in package.json to account for older Node.js versions - cc61b2c
* Reverted using `browser` field instead of `main` in package.json (for the UMD variant of the library): bundlers by default prefer it over `module` and will end up with a UMD module where ES may be preferred - cc61b2c

<hr /><em>This discussion was created from the release <a href='https://github.com/js-cookie/js-cookie/releases/tag/v3.0.0-rc.2'>v3.0.0-rc.2</a>.</em>

**Comment by proinc-bayles** [comment]
This release is broken when not loaded as a module. @carhartl
![image](https://user-images.githubusercontent.com/56029340/125681489-057324f2-72e5-44c3-8922-da0b84e575d3.png)

**Reply by emre42100** [reply]
<script src="https://cdnjs.cloudflare.com/ajax/libs/js-cookie/2.2.0/js.cookie.min.js"></script>

**Reply by Miniland1333** [reply]
Issue has been opened at #698

**Reply by proinc-bayles** [reply]
> ```
>         <script src="https://cdnjs.cloudflare.com/ajax/libs/js-cookie/2.2.0/js.cookie.min.js"></script>
> ```

Sure, but this likely broke the library in production for everyone who used the recommended script tag from the readme. It specifically says that it works without module support. There shouldn't be massive breaking changes from one rc to the next

**Reply by carhartl** [reply]
>this likely broke the library in production for everyone who used the recommended script tag from the readme

As the readme also suggests strongly at the top is to use the latest *stable* version. If you decide to use a beta or release candidate in production it's your responsibility. There might be breaking changes, be it by accident or not. That is why there are beta and release candidates to begin with - to denote potential instability, or bugs.

I agree nevertheless that there shouldn't be such breaking changes for release candidates. It was a mistake and mistakes happen, it wasn't done on purpose.
